kafka-streams/src/main/java/com/networknt/kafka/streams/LightStreams.java

```diff
@@ -30,4 +36,60 @@ default void registerModule() {
         masks.add("schema.registry.ssl.truststore.password");
         ModuleRegistry.registerModule(LightStreams.class.getName(), Config.getInstance().getJsonMapConfigNoCache(KafkaStreamsConfig.CONFIG_NAME), masks);
     }
+
+    default Object getKafkaValueByKey(ReadOnlyKeyValueStore<String, ?> keyValueStore, String key) {
+        Object returnObj = null;
+        boolean storeMoved;
+        long timeout = System.currentTimeMillis() + WAIT_THRESHOLD;
+        do {
+            if(System.currentTimeMillis() >= timeout) {
+                if(logger.isDebugEnabled()) logger.debug("Timeout period is passed after 30 seconds.");
+                break;
+            }
+            try {
+                returnObj = keyValueStore.get(key);
+                storeMoved = false;
+            } catch (InvalidStateStoreException e) {
+                storeMoved = true;
+                try {
+                    logger.debug(e.getMessage());
+                    Thread.sleep(100L);
+                } catch (InterruptedException ie) {
+                    logger.error(ie.getMessage(), ie);
+                }
+            }
+        } while (storeMoved);
+        if(returnObj == null) {
```

Comment:
I think this logging statement is miss leading. It might reach here immediately if the key is not in the current instance. Should we just remove it?